blueocean-core-js/src/js/ToastService.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * Created by cmeyers on 8/18/16.
+ */
+import { observable, computed } from 'mobx';
+
+/**
+ * Holds one or more toasts in state for display in UI.
+ */
+export class ToastService {
+
+    @observable toasts = [];
+
+    /**
+     * Creates a new toast that is added to the list.
+     *
+     * @param toast object with the following shape:
+     * {
+     *  text: string, message text to display
+     *  action: string, text for action link
+     *  onActionClick: function, callback to invoke when action link is clicked
+     *  onDismiss: function, callback to invoke when toast is dismissed (immediately, or after timeout)
+     *  dismissDelay: number, duration in millis after which to auto-dismiss this Toast
+     *  id: unique identifier (optional, will be autogenerated if ommitted)
+     * }
+     * @returns {number} unique ID of toast
+     */
+    newToast(toast) {
+        if (!toast.id) {
+            toast.id = Math.random() * Math.pow(10, 16);
+        }
+
+        // TODO: determine why it's necessary to re-set the "toasts" field to trigger the UI update
```

Comment:
With immutable that should not happen, I reckon react detects that you are passing around the same object and does not update UI for that reason. 

```
this.toasts = this.toasts.set(toast.id, toast)
```